app/models/game.rb

```diff
@@ -5,6 +5,87 @@ class Game < ActiveRecord::Base
 
   scope :needing_second_player, -> { where(black_user: nil) }
 
+  # find piece in game at x y coordinate
+  def find_piece(x, y)
+    pieces.where("x_position = ? AND y_position = ?", x, y)
+  end
 
+  # determine if the current square is occupied
+  def square_occupied?(x, y)
+    find_piece(x, y).any?
+  end
 
+  # determine if the total row/column/diagonal is occupied
+  def range_occupied?(x1, y1, x2, y2)
+    pieces.where("x_position BETWEEN ? AND ? AND y_position BETWEEN ? AND ?", x1, x2, y1, y2).any?
+  end
+
+  # is_obstructed? returns true if the current piece cannot move to its destination because a piece on its own team is in its way
+  def is_obstructed?(dest_x, dest_y)
+    # current location
+    current_x = x_position
```

Comment:
x_position and y_position won't exist on the Game model - you probably want this to either reference a piece in the arguments or maybe this should be inside the Piece model